analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ImplementSerializationMethodsCorrectly.CSharp9.cs

```diff
@@ -55,6 +56,38 @@ public record Foo
     [OnDeserialized()]
     private void OnDeserializedMethod(StreamingContext context) { }     // Compliant
 
+    [OnDeserialized]
+    [EditorBrowsable]
+    public static void OnDeserialized(StreamingContext context) { }      // Noncompliant {{Make this method non-public and non-static.}}
+
+    [OnDeserializing]
+    [EditorBrowsable]
+    public static void OnDeserializing(StreamingContext context) { }     // Noncompliant {{Make this method non-public and non-static.}}
+
+    [OnDeserialized]
+    [EditorBrowsable(EditorBrowsableState.Advanced)]
+    public static void OnDeserialized1(StreamingContext context) { }     // Noncompliant {{Make this method non-public and non-static.}}
+
+    [OnDeserializing]
+    [EditorBrowsable(EditorBrowsableState.Advanced)]
+    public static void OnDeserializing1(StreamingContext context) { }    // Noncompliant {{Make this method non-public and non-static.}}
+
+    [OnDeserialized]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public void OnDeserialized3(StreamingContext context) { }           // Compliant
+
+    [OnDeserializing]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public void OnDeserializing3(StreamingContext context) { }          // Compliant
+
+    [OnSerializing]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public void OnSerializing3(StreamingContext context) { }            // Compliant
+
+    [OnSerialized]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    int OnSerialized3(StreamingContext context) { return 1; }           // Compliant
+
```

Comment:
Do we need to duplicate the cases from `ImplementSerializationMethodsCorrectly.cs` aslo here?

Reply:
So basically in CSharp9 we want to test only C#9 specific syntax only, right?

Reply:
Yes